extensions/oidc/deployment/src/main/java/io/quarkus/oidc/deployment/OidcBuildStep.java

```diff
@@ -324,10 +323,12 @@ public SyntheticBeanBuildItem setup(
             OidcConfig config,
             OidcRecorder recorder,
             CoreVertxBuildItem vertxBuildItem,
-            TlsRegistryBuildItem tlsRegistryBuildItem) {
+            TlsRegistryBuildItem tlsRegistryBuildItem,
+            VertxWebRouterBuildItem routerBuildItem) {
```

Comment:
This leads to a `ChainBuildException` when runing the integration-tests:
```java
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:build (default) on project quarkus-integration-test-oidc: Failed to build quarkus application: io.quarkus.builder.ChainBuildException: Cycle detected:
[ERROR]                    io.quarkus.arc.deployment.ObserverValidationProcessor#validateApplicationObserver produced class io.quarkus.arc.deployment.ValidationPhaseBuildItem$ValidationErrorBuildItem
[ERROR]                 to io.quarkus.arc.deployment.ArcProcessor#generateResources produced class io.quarkus.arc.deployment.ResourcesGeneratedPhaseBuildItem
[ERROR]                 to io.quarkus.arc.deployment.ArcProcessor#initializeContainer produced class io.quarkus.arc.deployment.ArcContainerBuildItem
[ERROR]                 to io.quarkus.arc.deployment.ArcProcessor#notifyBeanContainerListeners produced class io.quarkus.arc.deployment.PreBeanContainerBuildItem
[ERROR]                 to io.quarkus.arc.deployment.ArcProcessor#signalBeanContainerReady produced class io.quarkus.arc.deployment.BeanContainerBuildItem
[ERROR]                 to io.quarkus.resteasy.reactive.server.deployment.ResteasyReactiveProcessor#setupDeployment produced class io.quarkus.vertx.http.deployment.RouteBuildItem
[ERROR]                 to io.quarkus.vertx.http.deployment.VertxHttpProcessor#initializeRouter produced class io.quarkus.vertx.http.deployment.VertxWebRouterBuildItem
[ERROR]                 to io.quarkus.oidc.deployment.OidcBuildStep#setup produced class io.quarkus.arc.deployment.SyntheticBeanBuildItem
[ERROR]                 to io.quarkus.arc.deployment.SyntheticBeansProcessor#initStatic produced class io.quarkus.arc.deployment.BeanRegistrationPhaseBuildItem$BeanConfiguratorBuildItem
[ERROR]                 to io.quarkus.arc.deployment.ArcProcessor#registerSyntheticObservers produced class io.quarkus.arc.deployment.UnremovableBeanBuildItem
[ERROR]                 to io.quarkus.arc.deployment.ArcProcessor#validate produced class io.quarkus.arc.deployment.ValidationPhaseBuildItem
[ERROR]                 to io.quarkus.arc.deployment.ObserverValidationProcessor#validateApplicationObserver
```
Since I'm new to the Quarkus BuildStep system, I don't know how to tackle this problem, any suggestions @sberyozkin ?

Reply:
I'm actually not sure, we'll need to ask next week Clement which item must be used to get hold of the main Router

Reply:
@sberyozkin Any news on which `BuildItem` to use for the main router in this case?

Reply:
> @sberyozkin Any news on which BuildItem to use for the main router in this case?

Use `io.quarkus.vertx.http.deployment.RouteBuildItem`.

Reply:
BTW none of these approaches will work anyway, you always need to know path during the build time, which you don't. Only method that would allow you to do it is deprecated for a long time now. Basically, #43046 is impossible to implement and current approach is OKish. I believe when someone requested to change it (Clement?) didn't realize that it is runtime property.

That's actually the reason why I left the form auth post location as build-time property https://github.com/quarkusio/quarkus/blob/2811eec1d8c1354332f354f1e4e444afc7b9d752/extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/HttpSecurityProcessor.java#L125

Reply:
I can also put it differently. If you use `VertxWebRouterBuildItem` what you gain is that it happens between the `VertxWebRouterBuildItem` is produced and the CDI event that is now consumed (router is observed). You get to register it sooner, but it still happens before app started app in both cases. I don't understand what would be gain here.

Anyway, if you really want to use the `VertxWebRouterBuildItem` you need to break the circular dependency. So in your case, I think you need to consume this build item before the step that produces the synthetic beans happen.